types/fs-base-ui/html_ui/JS/common.d.ts

```diff
@@ -0,0 +1,1102 @@
+         * @param withMinutes Whether or not the return value should contain minutes, when false also excludes seconds,
+         * e.g. 3740 seconds becomes: "01".
+         * @param withSeconds Whether or not the return value should contain seconds, when false 3740 seconds
+         * becomes: "01:02".
+         * @param doLocalize Whether or not to localize the string, defaults to true.
+         */
+        function SecondsToDisplayDuration(totalSeconds: number, withMinutes: boolean, withSeconds: boolean,
+                                          doLocalize?: boolean): string;
+
+        /**
+         * Converts the given totalSeconds to a time string. Currently it is unknown how this differs from
+         * SecondsToDisplayDuration.
+         * @param totalSeconds The total number of seconds to calculate a time string for.
+         * @param withMinutes Whether or not the return value should contain minutes, when false also excludes seconds,
+         * e.g. 3740 seconds becomes: "01".
+         * @param withSeconds Whether or not the return value should contain seconds, when false 3740 seconds
+         * becomes: "01:02".
+         * @param doLocalize Whether or not to localize the string, defaults to true.
+         */
+        function SecondsToDisplayTime(totalSeconds: number, withMinutes: boolean, withSeconds: boolean,
+                                      doLocalize?: boolean): string;
+
+        /**
+         * Converts the given hours, minutes and seconds to a time string: "hh:mm:ss".
+         * When minutes < 0: "hh". When seconds < 0: "hh:mm".
+         */
+        function timeToString(hours: number, minutes: number, seconds: number): string;
+
+        /**
+         * Determines whether a file by the given fileName exists.
+         */
+        function doesFileExist(fileName: string): boolean;
+
+        /**
+         * Loads the contents of the given file and provides it through the successCallback.
+         */
+        function loadFile(fileName: string, successCallback: (contents: string) => void): void;
+
+        /**
+         * Clones the given object using JSON stringify and parse.
+         */
+        function slowDeepClone(object: any): any;
+
+        function showTooltip(id: string, tooltip: string, posXRel: number, posYRel: number, maxWidth?: number): void;
+
+        function hideTooltip(id: string): void;
+
+        /**
+         * Adds leading (and trailing) zeros to the given value.
+         * @param value The numeric value to add zeros to.
+         * @param nbDigits The minimum number of digits before the decimal separator.
+         * @param pointFixed The minimum number of digits after the decimal separator.
+         */
+        function leadingZeros(value: number, nbDigits: number, pointFixed?: number): string;
+
+        /**
+         * Counts the number of digits after the decimal separator.
+         */
+        function countDecimals(value: number): number;
```

Comment:
Would also work if `value` already was, say, a string.

Reply:
Same as my earlier comment: I'll add string to the types. I don't think adding any other type would be in line with the "thought" behind this function.